src/Mod/PartDesign/App/FeatureRevolution.cpp

```diff
@@ -161,7 +157,14 @@ App::DocumentObjectExecReturn* Revolution::execute()
 
         // Create a fresh support even when base exists so that it can be used for patterns
         TopoShape result(0);
-        TopoShape supportface = getSupportFace();
+        TopoShape supportface(0);
+        try {
+            supportface = getSupportFace();
+        }
+        catch(...) {
+            //do nothing, null shape is handle below

```

Comment:
What is the try/catch block good for? In case getSupportFace() throws an exception then supportface is a null shape but unlike to your comment there is no special handling of null shapes in the code below.

Reply:
Sorry for the delay, I was away.
Line 186 we have
```cpp
            TopExp_Explorer Ex(supportface.getShape(), TopAbs_WIRE);
            if (!Ex.More()) {
                supportface = TopoDS_Face();
            }
```
which initialize `supportface` as a null face. This null face is good enough for `BRepFeat_MakeRevol::Init` called in `TopoShape::makeElementRevolution`.

Reply:
OK.